airgun/views/contentview_new.py

```diff
@@ -56,3 +88,148 @@ def is_displayed(self):
+
+        def search(self, version_name):
+            """Searches for content view version.
+            Searchbox can't search by version name, only by id, that's why in
+            case version name was passed, it's transformed into recognizable
+            value before filling, for example::
+                'Version 1.0' -> 'version = 1'
+            """
+            search_phrase = version_name
+            if version_name.startswith('V') and '.' in version_name:
+                search_phrase = f'version = {version_name.split()[1].split(".")[0]}'
+            self.searchbox.search(search_phrase)
+            return self.table.read()
+
+    @View.nested
+    class content_views(Tab):
+        TAB_LOCATOR = ParametrizedLocator('//a[contains(@href, "#/contentviews")]')
+
+        resources = View.nested(NewAddRemoveResourcesView)
+
+    @View.nested
+    class repositories(Tab):
+        TAB_LOCATOR = ParametrizedLocator('//a[contains(@href, "#/repositories")]')
+        resources = View.nested(NewAddRemoveResourcesView)
+
+    @View.nested
+    class filters(Tab):
+        TAB_LOCATOR = ParametrizedLocator('//a[contains(@href, "#/filters")]')
+        new_filter = Text(".//button[@ui-sref='content-view.yum.filters.new']")
+
+
+class ContentViewVersionPublishView(BaseLoggedInView):
+    # publishing view is a popup so adding all navigation within the same context
+    breadcrumb = BreadCrumb()
+    ROOT = './/div[contains(@class,"pf-c-wizard")]'
+    title = Text("//h2[contains(., 'Publish' or contains(@id, 'pf-wizard-title-0')]")
+    # publishing screen
+    description = TextInput(id='description')
+    promote = Switch('promote-switch')
+
+    # review screen only has info to review
+    # shared buttons at bottom for popup for both push and review section
+    next = Button('Next')
+    finish = Button('Finish')
+    back = Button('Back')
+    cancel = Button('Cancel')
+    close_button = Button('Close')
+    progressbar = PF4ProgressBar()
+
+    @property
+    def is_displayed(self):
+        breadcrumb_loaded = self.browser.wait_for_element(self.breadcrumb, exception=False)
+        return (
+            breadcrumb_loaded
+            and self.breadcrumb.locations[0] == 'Content Views'
+            and self.breadcrumb.read() == 'Versions'
+        )
+
+    def wait_animation_end(self):
+        wait_for(
+            lambda: 'in' in self.browser.classes(self),
+            handle_exception=True,
+            logger=self.logger,
+            timeout=10,
+        )
+
+    def before_fill(self, values=None):
+        """If we don't want to break view.fill() procedure flow, we need to
+        push 'Edit' button to open necessary dialog to be able to fill values
+        """
+        self.promote.click()
+        wait_for(
+            lambda: self.lce.is_displayed is True,
+            timeout=30,
+            delay=1,
+            logger=self.logger,
+        )
+
+
+class NewContentViewVersionDetailsView(BaseLoggedInView):
```

Comment:
Please add doc string for a class which would help to understand added functionality.